cli/azd/internal/vsrpc/server.go

```diff
@@ -115,21 +135,47 @@ func serveRpc(w http.ResponseWriter, r *http.Request, handlers map[string]Handle
 		}
 
 		go func() {
+			var respErr error
+			childCtx, span := tracing.Start(ctx, events.VsRpcEventPrefix+req.Method())
+			span.SetAttributes(fields.RpcMethod.String(req.Method()))
+			defer func() {
+				if respErr != nil {
+					cmd.MapError(respErr, span)
+					var rpcErr *jsonrpc2.Error
+					if errors.As(respErr, &rpcErr) {
+						span.SetAttributes(fields.JsonRpcErrorCode.Int(int(rpcErr.Code)))
+					}
+				}
+
+				// Include any usage attributes set
+				span.SetAttributes(tracing.GetUsageAttributes()...)
+				span.End()
+			}()
+
+			// Wrap the reply function to capture the response error returned by the handler before replying.
+			origReply := reply
+			reply = func(ctx context.Context, result interface{}, err error) error {
+				if err != nil {
+					respErr = err
+				}
+				return origReply(ctx, result, err)
+			}
+
```

Comment:
Does the go server support any lifetime hooks / middleware support similar where we can register these type of per request cross cutting concerns?

Reply:
The `Handler` interface would lend itself well to implementing composed functions (i.e., Middleware). For now, having a clear picture of the entire request handling flow may be more practical.